### PR TITLE
Minor refactor of the spectrum solver

### PR DESCRIPTION
**Pull request by andrewfullard**
### :pencil: Description

**Type:** :roller_coaster: `infrastructure`

Adjustment to the spectrum solver to consolidate property assignments and factor out luminosity integration calculations (which do not use anything relevant from the spectrum solver). `setup_optional_spectra` is may be subsumed into `solve` eventually.

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [x] I requested two reviewers for this pull request
- [x] I updated the documentation according to my changes
- [x] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: tardis/spectrum/luminosity.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import astropy.units as u
+import numpy as np
+
+
+def calculate_emitted_luminosity(
```
Comment (andrewfullard) [inline tardis/spectrum/luminosity.py:5]: Could maybe become a solver class if we want to be extra about it.

Comment (Rodot-) [inline tardis/spectrum/luminosity.py:5]: Are these covered by tests?

**Review by Rodot- — CHANGES_REQUESTED** [review]
I think tests for these functions are needed, will approve once added

**Review by Rodot- — APPROVED** [review]
This all looks good to merge for me

**Review by wkerzendorf — APPROVED** [review]

**Pull request merged**